packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,591 @@
+        }()),
+        super(key: key);
+
+  /// The list of widgets used to populate the tabs of the bar.
+  final List<Widget> children;
+
+  /// The [Key] value of the widget that should be initially selected.
+  ///
+  /// This [Key] must be the [Key] of one of the widgets within [children].
+  final Key value;
+
+  /// The callback called when an option is pressed.
+  ///
+  /// This callback should set the current [value] equal to the returned [Key].
+  final ValueChanged<Key> onValueChanged;
+
+  @override
+  _SegmentedControlState createState() => _SegmentedControlState();
+}
+
+class _SegmentedControlState extends State<SegmentedControl> {
+  Key _selectedValue;
+  Key _pressedValue;
+  final List<Widget> _gestureChildren = <Widget>[];
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedValue = widget.value;
+  }
+
+  void _onTapDown(int index) {
+    setState(() {
+      _pressedValue = widget.children.elementAt(index).key;
+    });
+  }
+
+  void _onTapUp(int index) {
+    setState(() {
+      _pressedValue = null;
+    });
+  }
+
+  void _onTapCancel(int index) {
+    setState(() {
+      _pressedValue = null;
+    });
+  }
+
+  void _onTap(int index) {
+    setState(() {
+      _selectedValue = widget.children.elementAt(index).key;
+      widget.onValueChanged(_selectedValue);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _gestureChildren.clear();
+    for (int i = 0; i < widget.children.length; i += 1) {
+      final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+            color: (_selectedValue == widget.children.elementAt(i).key)
+                ? CupertinoColors.white
+                : CupertinoColors.activeBlue,
+          );
+      final IconThemeData iconTheme = IconThemeData(
+        color: (_selectedValue == widget.children.elementAt(i).key)
+            ? CupertinoColors.white
+            : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children.elementAt(i);
+      child = GestureDetector(
+          onTapDown: (TapDownDetails event) {
+            _onTapDown(i);
+          },
+          onTapUp: (TapUpDetails event) {
+            _onTapUp(i);
+          },
+          onTapCancel: () {
+            _onTapCancel(i);
+          },
+          onTap: () {
+            _onTap(i);
+          },
+          child: new IconTheme(
+              data: iconTheme,
+              child: DefaultTextStyle(
+                style: textStyle,
+                child: child,
+              )),
+          key: widget.children.elementAt(i).key);
+      _gestureChildren.add(child);
+    }
+
+    final Widget box = new _SegmentedControlRenderWidget(
+      children: _gestureChildren,
+      selectedValue: _selectedValue,
+      pressedValue: _pressedValue,
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(constrainedAxis: Axis.horizontal, child: box,),
```

Comment:
You still need to reformat this so that the comma at the end forces dartfmt to put these arguments on separate lines.